### PR TITLE
Support HTML5 Drag and drop (closes #897)

### LINKED ISSUE #897: Support HTML5 Drag and drop
### Are you requesting a feature or reporting a bug?

feature
### Specify your
- operating system: windows 7
- testcafe version: 0.9.0
- node.js version: 6.2.1

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

File: src/client/automation/playback/move/index.js [hunk 1]
```diff
@@ -301,7 +294,8 @@ export default class MoveAutomation {
             .move(this.x, this.y)
             .then(getElementUnderCursor)
             // NOTE: in touch mode, events are simulated for the element for which mousedown was simulated (GH-372)
-            .then(topElement => this._emulateEvents(this.dragMode && this.touchMode ? this.dragElement : topElement))
+            .then(topElement => this._emulateEvents(this.holdLeftButton &&
```
Comment (helen-dikareva) [inline src/client/automation/playback/move/index.js]: `{}` is required

File: src/client/automation/playback/move/index.js [hunk 2]
```diff
@@ -387,7 +381,34 @@ export default class MoveAutomation {
     run () {
         return getElementUnderCursor()
             .then(topElement => {
-                this.dragElement = this.dragMode ? topElement : null;
+                this.dragElement = this.holdLeftButton ? topElement : null;
+
+                var parentNode = this.dragElement;
+
+                while (parentNode && !this.dragAndDropMode) {
```
Comment (helen-dikareva) [inline src/client/automation/playback/move/index.js]: may better to extract this in method

File: src/client/automation/playback/drag/base.js [hunk 3]
```diff
@@ -36,9 +36,13 @@ export default class DragAutomationBase {
 
         this.automationSettings = new AutomationSettings(this.speed);
 
-        this.endPoint  = null;
-        this.downEvent = browserUtils.isTouchDevice ? 'touchstart' : 'mousedown';
-        this.upEvent   = browserUtils.isTouchDevice ? 'touchend' : 'mouseup';
+        this.endPoint        = null;
+        this.downEvent       = browserUtils.isTouchDevice ? 'touchstart' : 'mousedown';
+        this.upEvent         = browserUtils.isTouchDevice ? 'touchend' : 'mouseup';
+        this.dragAndDropMode = false;
```
Comment (helen-dikareva) [inline src/client/automation/playback/drag/base.js]: may be combine options required for drag&drop mode in one objest, smth like `this.dragAndDropState = {element: null, ...}`

File: src/client/automation/playback/move/event-sequence/base.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import { domUtils } from '../../../deps/testcafe-core';
+
+var browserUtils = hammerhead.utils.browser;
+
+export default class MoveEventSequenceBase {
+    constructor () {
+    }
+
+    setup () {
+        this.dragAndDropMode = false;
```
Comment (helen-dikareva) [inline src/client/automation/playback/move/event-sequence/base.js]: I think properties should be assigned in constructor

File: src/client/automation/playback/move/event-sequence/first-drag-and-first-move.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import hammerhead from '../../../deps/hammerhead';
+import MoveEventSequenceBase from './base';
+import moveEventSequence from './index';
+import dragAndDropMoveEventSequence from './drag-and-drop-move';
+
+var eventSimulator = hammerhead.eventSandbox.eventSimulator;
+
+class DragAndDropFirstMoveEventSequence extends MoveEventSequenceBase {
```
Comment (helen-dikareva) [inline src/client/automation/playback/move/event-sequence/first-drag-and-first-move.js]: I think file name should be `drag-and-drop-first-move`

**Comment by churkin** [comment]
rebase required

**Comment by AlexanderMoskovkin** [comment]
rebased, waiting for the tests

**Comment by AlexanderMoskovkin** [comment]
FPR
/cc @churkin

**Pull request merged**